tobs

```diff
@@ -534,14 +534,22 @@ sub update_spec
     }
 
     if(/^Name:(\s*)(\S+)/) {
-      $spec_name = $2;
+      # Be careful: if there are several 'Name:' lines don't do fancy stuff
+      # as we are not sure which one is correct (think of %if's in the
+      # spec).
+      if(defined $spec_name) {
+        $spec_name = '';
+      }
+      else {
+        $spec_name = $2;
+      }
     }
 
     if(/^Source:(\s*)((\S+)\.tar\.(bz2|gz|xz))/) {
       $config->{rm_archive} = $2;
       my $s = $1;
       my $n = $config->{archive};
-      $config->{rm_archive} =~ s/%\{name\}/$spec_name/g;
+      $config->{rm_archive} =~ s/%\{name\}/$spec_name/g if $spec_name ne '';
```

Comment:
so what happens if there are more and `%{name}` is defined? it tries to remove `%{name}-%{version}.tar.bz2` which does not exists?

Reply:
Yes. You better not use %{name} in package source names, then.

This is nearly a theoretical problem but `installation-images` has historically really two `Name:` lines.